v2 docs for ScaledObject (#211)
kedacore/keda-docs: content/docs/2.0/concepts/scaling-deployments.md

  name: {scaled-object-name}
spec:
  scaleTargetRef:
    apiVersion:    {api-version-of-target-resource}  # Optional. Default: apps/v1
    kind:          {kind-of-target-resource}         # Optional. Default: Deployment
    name:          {name-of-target-resource}         # Mandatory. Must be in the same namespace as the ScaledObject
    containerName: {container-name}                  # Optional. Default: .spec.template.spec.containers[0]
  pollingInterval: 30                                # Optional. Default: 30 seconds
  cooldownPeriod:  300                               # Optional. Default: 300 seconds
  minReplicaCount: 0                                 # Optional. Default: 0
  maxReplicaCount: 100                               # Optional. Default: 100
  advanced:                                          # Optional. Section to specify advanced options.
    horizontalPodAutoscalerConfig:                   # Optional. If not set, KEDA won't scale based on resource utilization
      resourceMetrics:

>       - name: cpu/memory                             # Name of the resource to be targeted

Reviewer: What about this?
```suggestion
      - name: cpu/memory                             # Name of the metric to scale on
```

Author: Good catch, makes sense, will fix that

  # {list of triggers to activate scaling of the target resource}
```

You can find all supported triggers [here](/scalers).

### Details
```yaml
  scaleTargetRef:
    apiVersion:    {api-version-of-target-resource}  # Optional. Default: apps/v1
    kind:          {kind-of-target-resource}         # Optional. Default: Deployment
    name:          {name-of-target-resource}         # Mandatory. Must be in the same namespace as the ScaledObject
    containerName: {container-name}                  # Optional. Default: .spec.template.spec.containers[0]
```

The reference to the resource this ScaledObject is configured for. This is the resource KEDA will scale up/down and setup an HPA for, based on the triggers defined in `triggers:`. 

To scale Kubernetes Deployments only `name` is needed to be specified, if one wants to scale a different resource such as StatefulSet or  Custom Resource (that defines `/scale` subresource), appropriate `apiVersion` (following standard Kubernetes convetion, ie. `{api}/{version}`) and `kind` need to be specfied.

`containerName` is a name of container in the target resource, from which KEDA should try to get environment properties holding secrets etc.

> **Assumptions:** Resource referenced by `name` (and `apiVersion`, `kind`) is in the same namespace as the scaledObject

Reviewer: Should we put it like this or be explicit and say we only support scaling workloads in the same namespace as the SO?

Author: What do you propose by explicit?

### Details
```yaml
  scaleTargetRef:
    apiVersion:    {api-version-of-target-resource}  # Optional. Default: apps/v1
    kind:          {kind-of-target-resource}         # Optional. Default: Deployment

>     name:          {name-of-target-resource}         # Must be in the same namespace as the ScaledObject

Reviewer: ```suggestion
    name:          {name-of-target-resource}         # Mandatory. Must be in the same namespace as the ScaledObject
```
